synapse/lib/stormlib/model.py

```diff
@@ -1269,3 +1020,269 @@ async def _riskHasVulnToVulnerable(self, n, nodata=False):
+
+            for tagname, tagvalu in sode.get('tags', {}).items():
+                if tagvalu == (None, None):
+                    await self.runt.printf(f'    #{tagname}')
+                else:
+                    mintime, maxtime = tagvalu
+                    mindt = s_time.repr(mintime)
+                    maxdt = s_time.repr(maxtime)
+                    await self.runt.printf(f'    #{tagname} = ({mindt}, {maxdt})')
+
+            for tagprop, tagpropvalu in sode.get('tagprops', {}).items():
+                for prop, valu in tagpropvalu.items():
+                    await self.runt.printf(f'    #{tagprop}:{prop} = {valu[0]}')
+
+        if sources := node['sources']:
+            await self.runt.printf(f'  sources: {sorted(sources)}')
+
+        if noderefs := node['refs']:
+            await self.runt.printf('  refs:')
+
+            for layriden, reflist in noderefs.items():
+                await self.runt.printf(f'    layer: {layriden}')
+                for iden, refinfo in reflist:
+                    form, prop, *_ = refinfo
+                    await self.runt.printf(f'      - {form}:{prop} (iden: {iden}')
+
+        n1edges = node['n1edges']
+        n2edges = node['n2edges']
+
+        if n1edges or n2edges:
+            await self.runt.printf('  edges:')
+
+        for layriden, edges in n1edges.items():
+            for verb, iden in edges:
+                await self.runt.printf(f'    -({verb})> {iden}')
+
+        for layriden, edges in n2edges.items():
+            for verb, iden in edges:
+                await self.runt.printf(f'    <({verb})- {iden}')
+
+    async def _removeNode(self, offset):
+        offset = await s_stormtypes.toint(offset)
+        await self.runt.snap.core.coreQueuePop('model_0_2_31:nodes', offset)
+
+    async def _repairNode(self, offset, newvalu):
+
+        item = await self.runt.snap.core.coreQueueGet('model_0_2_31:nodes', offset, cull=False)
+        if item is None:
+            await self.runt.warn(f'Queued node with offset {offset} not found.')
+            return False
+
+        node = item[1]
+
+        nodeform = node['formname']
+        form = self.runt.snap.core.model.form(nodeform)
+
+        norm, info = form.type.norm(newvalu)
+
+        for layriden in node['layers']:
+            layer = self.runt.snap.core.getLayer(layriden)
+            if layer is None: # pragma: no cover
+                await self.runt.warn(f'Layer does not exist to recreate node: {layriden}.')
+                return False
+
+        await self.runt.printf(f'Repairing node at offset {offset} from {node["formvalu"]} -> {norm}')
+
+        buid = s_common.buid((nodeform, norm))
+
+        nodeedits = []
+
+        # Create the node in the right layers
+        for layriden in node['layers']:
+            nodeedits.append((layriden, (
+                (buid, nodeform, (
+                    (s_layer.EDIT_NODE_ADD, (norm, form.type.stortype), ()),
+                )),
```

Comment:
Do you think it would be worth it to combine the edits for each node rather than having it be a bunch of single edits?

Reply:
Applying node edits is pretty quick, especially in comparison to iterating all the layers/nodes. If you think it's worth it, this should be a pretty easy change to `_flushEdits()`.

Reply:
I'll bring it up as a goback

Reply:
Done.